Update Prefix-Cache-Scorer `cache_tracking` Mode with The v0.2 KVCache.Indexer (#245)
llm-d/llm-d-inference-scheduler: docs/architecture.md

If the configuration is in a file, the EPP command line argument `--configFile` should be used to specify the full path of the file in question. If the configuration is passed as in-line text the EPP command
line argument `--configText` should be used.

---

### Plugin Configuration

This section describes how to setup the various plugins available with the llm-d-inference-scheduler

#### PrefillHeader

Sets a header for use in disaggregated prefill/decode

- **Type**: `prefill-header-handler`
- **Parameters**:

>   - `prefillProfile`: specifies the name of the profile used for the prefill scheduling. Only needed if the prefill profile is not named `prefill`.

Reviewer: ```suggestion
  - `prefillProfile`: specifies the name of the profile used for the prefill scheduling. Defaults to `prefill`.
```

---

#### PdProfileHandler

Selects the profiles to use when running with disagregated prefill/decode

- **Type**: `pd-profile-handler`
- **Parameters**:
  - `threshold`: specifies the threshold at which there are enough new input tokens to send the request to prefill and then decode, vs just to decode.

>   - `hashBlockSize`: specifies the length of the prompt chunk that a block is keyed by. This must the same value used for the PrefixCachePlugin.

Reviewer: ok for now. for the long run it doesn't make sense to define same hashBlockSize in two places.
we might need to either add a parameters sections that all plugins can consume selected params, or change implementation of pd profile handler.

Reviewer 2: Agreed. Again this is just a refactoring of the docs formatting, the extended parts are in the `prefix-cache-scorer` `cache_tracking` mode. Sorry for the confusion.

>   - `decodeProfile`: specifies the name of the profile used for the decode scheduling. Only needed if the decode profile is not named `decode`.

Reviewer 2: See above comment on docs.

  - `maxPrefixBlocksToMatch`: Specifies the maximum number of prefix blocks to match. Defaults to `256` if not specified.
  - `lruCapacityPerServer`: Specifies the capacity of the LRU indexer, in number of entries per server (pod). Defaults to `31,250` if not specified.

**Note:** `mode: estimate` is not required, as it is the default.

##### `cache_tracking` mode:

This mode scores requests based on the actual KV-cache states across the vLLM instances. 
It is more accurate than both `SessionAffinity` and `PrefixCachePlugin` in `estimate` mode,
but incurs additional computation overhead and KV-Events streaming to track the KV-cache states.

When enabled, the scorer will use the `llm-d-kv-cache-manager` to track the KV-cache states across the vLLM instances.
It will use the `kvcache.Indexer` to score the pods based on the number of matching blocks in the KV-cache.
It will also use the `kvevents.Pool` to subscribe to the KV-Events emitted by the vLLM instances and update the KV-cache states in near-real-time.

Configuration:

- **Type**: `prefix-cache-scorer`
- **Parameters**:
  - `mode: cache_tracking`

>   - `indexerConfig`: Configuration for the `kvcache.Indexer`.

Reviewer 3: I think you should specify it her,e the value,s at least the important ones

Reviewer 2: They are specified in lines under
